src/tests/system/tests/test_sssctl.py

```diff
@@ -7,279 +7,135 @@
+    "pattern,repl",
     [
-        pytest.param(
-            "id_provider.*",
-            "id_provider = invalid",
-            "Attribute 'id_provider' in section 'domain/local' has an invalid value: invalid",
-            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
-        ),
-        pytest.param(
-            "id_provider.*",
-            "",
-            "Attribute 'id_provider' is missing in section 'domain/local'.",
-            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
-        ),
-        ("id_provider", "id_@provider", "Attribute 'id_@provider' is not allowed in section"),
-        ("domain/local", "domain/local@", "Section [domain/local@] is not allowed"),
-        (".sssd.", "[sssdx]", "Section [sssdx] is not allowed"),
+        ("id_provider.*", "id_provider = invalid"),
+        ("id_provider.*", ""),
+        ("id_provider", "id_@provider"),
+        ("domain/local", "domain/local@"),
+        (".sssd.", "[sssdx]"),
+        ("id_provider = ", "id_provider "),
+        (".nss.", "[nssx"),
+        (".domain/local.", "domain/local]"),
     ],
 )
 @pytest.mark.tools
 @pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_invalid_semantic_in_section_name(client: Client, pattern: str, repl: str, expected: str):
+def test_sssctl__invalid_config_patterns(client: Client, pattern: str, repl: str):
     """
-    :title: sssctl prints appropriate error message if semantic in section name is invalid
+    :title: sssctl config-check detects invalid config patterns
     :setup:
-        1. Start SSSD
-        2. Edit sssd.conf with invalid semantic in section name
+        1. Start with valid SSSD config
+        2. Modify config with invalid patterns
     :steps:
-        1. Validate configuration file using sssctl config-check
+        1. Run sssctl config-check
     :expectedresults:
-        1. sssctl configuration check fails with the correct output in stdout
+        1. Configuration check fails for all invalid patterns
     :customerscenario: False
     """
     client.sssd.common.local()
     client.sssd.start()
     conf = re.sub(pattern, repl, client.fs.read("/etc/sssd/sssd.conf"))
     client.fs.write("/etc/sssd/sssd.conf", conf)
-
     result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config!"
-    assert expected in result.stdout, "Wrong error message on stdout!"
+    assert result.rc != 0, f"config-check should fail for pattern '{pattern}' -> '{repl}'!"
 
 
 @pytest.mark.parametrize(
-    "pattern,repl,expected",
+    "scenario, action",
     [
-        ("id_provider = ", "id_provider ", "Equal sign is missing"),
-        (".nss.", "[nssx", "No closing bracket"),
-        (".domain/local.", "domain/local]", "Equal sign is missing"),
+        ("invalid_permissions", lambda c: c.fs.chmod("0777", "/etc/sssd/sssd.conf")),
+        ("missing_config", lambda c: c.fs.rm("/etc/sssd/sssd.conf")),
```

Comment:
These should be just to testcases parametrziation make it just unreadable and is overkill there.